loki/transformations/single_column/vector.py

```diff
@@ -141,7 +141,11 @@ def extract_vector_sections(cls, section, horizontal):
                 # as 'Conditional's rely on the fact that the first element of the 'else_body'
                 # (if 'has_elseif') is a Conditional itself
                 if separator.has_elseif and separator.else_body:
```

Comment:
Since `separator.else_bodies` should, even in the `has_elseif==False` case, return the else_body correctly, we should be able to remove this check here and always use the new implementation, no?